src/common/survival_util.h

```diff
@@ -0,0 +1,103 @@
+/*!
+ * Copyright 2019 by Contributors
+ * \file survival_util.h
+ * \brief Utility functions, useful for implementing objective and metric functions for survival
+ *        analysis
+ * \author Avinash Barnwal, Hyunsu Cho and Toby Hocking
+ */
+#ifndef XGBOOST_COMMON_SURVIVAL_UTIL_H_
+#define XGBOOST_COMMON_SURVIVAL_UTIL_H_
+
+#include <xgboost/enum_class_param.h>
+#include <memory>
+
+namespace xgboost {
+namespace common {
+
+// Choice of distribution for the noise term in AFT
+enum class AFTDistributionType : int {
+  kNormal = 0, kLogistic = 1, kExtreme = 2
+};
+
+}  // namespace common
+}  // namespace xgboost
+
+DECLARE_FIELD_ENUM_CLASS(xgboost::common::AFTDistributionType);
+
+namespace xgboost {
+namespace common {
+
+// Constant PI
+const double kPI = 3.14159265358979323846;
```

Comment:
I think libc has a `M_PI` defined.

Reply:
`M_PI` is not part of standard C++. Only GLIBC defines it.